File: examples/input/templates/logger.ts
import { Client } from 'pg';
// #BEGIN_IMPORT
// #IMPORT2 import {#IMPORT_1} from '{#IMPORT_2}';
// #END_IMPORT

export default class {#NAME_PASCAL} {
  public static async log(
// #BEGIN_VAR
    // #VAR {#VARNAME_CAMEL}: {#TYPE} = {#TYPE_INIT},
// #END_VAR
  ): Promise<boolean> {
    const client = new Client();
    client.connect();

    const sqlStatement = 'INSERT INTO {#NAME_PASCAL} ({#VARNAME_SNAKE_LIST}) VALUES({#VARNAME_COUNT_LIST})';
    const values = [{#VARNAME_CAMEL_LIST}];

    try {
      await client.query(sqlStatement, values);
      return true;
    } catch (err) {
      return false;
    }
  }
}

File: examples/input/templates/logger.go
package {#1}
// #BEGIN_IMPORT

import (
	// #IMPORT2 {#IMPORT_1} "{#IMPORT_2}"
	// #IMPORT1 "{#IMPORT_1}"
)
// #END_IMPORT
// #BEGIN_VAR

func Log{#NAME_PASCAL}(
	// #VAR {#VARNAME_LOWER} {#TYPE},
) bool {
  sqlStatement := `
    INSERT INTO {#NAME_PASCAL} ({#VARNAME_SNAKE_LIST})
    VALUES ({#VARNAME_COUNT_LIST})
  `
  err := db.QueryRow(sqlStatement, {#VARNAME_LOWER_LIST})
  return err != nil
}
// #END_VAR
// #BEGIN_FUNCTIONS

// #FUNCTIONS {#FUNCTIONS}
// #END_FUNCTIONS
